documentation/CAMARA-Security-Interoperability.md

```diff
@@ -0,0 +1,158 @@
+
+## Client Authentication
+
+This CAMARA document allows **one** client authentication method, private_key_jwt, as defined in OIDC
+[OIDC Client Authentication](https://openid.net/specs/openid-connect-core-1_0.html#ClientAuthentication)
+[JSON Web Token (JWT) Profile for OAuth 2.0 Client Authentication and Authorization Grants](https://www.rfc-editor.org/rfc/rfc7523.html#section-2.2)
+
+## Sender-constraint Access Token
+
+This CAMARA document RECOMMENDS sender-constraint access tokens.
+
+### DPoP Requirements for Camara Clients
+
+Camara clients SHOULD utilize DPoP as described in [RFC9449].
+
+If Camara clients use DPoP, they shall support the server provided nonce mechanism (as defined in Section 8 of [RFC9449])
+
+### DPoP Requirements for Camara Authorizatin Servers
+
+Camara Authorization Servers MUST support DPoP as described in [RFC9449].
+
+### DPoP Requirements for Resource Servers
+
+Camara Resource Servers MUST support and verify sender-constrained access tokens using DPoP as described in [RFC9449].
+
+## OpenId Foundation Certification
+
+Camara recommends that implementations run the OIDF interoperability suite and achive [OIDF certification](https://openid.net/certification/).
+
+## References
+
+* [Data Privacy Vocabulary (DPV)](https://w3c.github.io/dpv/dpv/)
+* [E.164 - The international public telecommunication numbering plan](https://www.itu.int/rec/T-REC-E.164-201011-I/en)
+* [OpenID Connect Core 1.0 specification](https://openid.net/specs/openid-connect-core-1_0.html)
+* [OpenID Connect Client-Initiated Backchannel Authentication Flow - Core 1.0](https://openid.net/specs/openid-client-initiated-backchannel-authentication-core-1_0.html)
+* [RFC 2617 - HTTP Authentication: Basic and Digest Access Authentication](https://www.rfc-editor.org/info/rfc2617)
+* [RFC 3986 - Uniform Resource Identifier (URI): Generic Syntax](https://www.rfc-editor.org/info/rfc3986)
+* [RFC 3966 - The tel URI for Telephone Numbers](https://www.rfc-editor.org/info/rfc3966)
+* [RFC 6749 - The OAuth 2.0 Authorization Framework](https://www.rfc-editor.org/info/rfc6749)
+* [RFC 7515 - JSON Web Signature (JWS)](https://www.rfc-editor.org/info/rfc7515)
+* [RFC 7519 - JSON Web Token (JWT)](https://www.rfc-editor.org/info/rfc7519)
+* [RFC 7521 - Assertion Framework for OAuth 2.0 Client Authentication and Authorization Grants](https://www.rfc-editor.org/info/rfc7521)
+* [RFC 7523 - JSON Web Token (JWT) Profile for OAuth 2.0 Client Authentication and Authorization Grants](https://www.rfc-editor.org/info/rfc7523)
+* [RFC 8259 - The JavaScript Object Notation (JSON) Data Interchange Format](https://www.rfc-editor.org/info/rfc8259)
+* [RFC 8414 - OAuth 2.0 Authorization Server Metadata](https://www.rfc-editor.org/info/rfc8414)
+* [RFC 9449 - OAuth 2.0 Demonstrating Proof of Possession (DPoP)](https://www.rfc-editor.org/info/rfc9449)
+
```

Comment:
This PR barely adds anything over the current situation. Simply saying "read and implement the RFC" sounds good in theory. However, that is the current strategy and has proven to be problematic in practice (this is why we decided we need a profile in the first place).

To illustrate it with two recent cases:

- The RFC states, "OpenID Connect requests MUST contain the **openid scope** value. If the openid scope value is not present, the behavior is entirely unspecified." This lack of clarity led to issues in OpenGateway because one CSP in Europe argued that the text is open to not sending the scope. The result: some telcos are not interoperable due to differing interpretations, and the aggregator must adapt the requests to the telco using the less-restrictive interpretation. To address this, #113 proposes a more detailed profile stating, "OpenID Connect requests MUST contain the openid scope value. If the openid scope value is not present, the ID Token will not be returned," leaving no room for interpretation.
- Another issue concerns the usage of **acr_values**. According to the current profile, sending them is optional. However, at least two telcos in OpenGateway are using different values, causing interoperability issues. Standardizing the possible acr values is necessary to ensure developers can build apps that work across different telcos. This is why #113 does not include that field (until we standardize it if needed).

These are real examples, not fabricated. I'm omiting the telco names, but they are represented in this working group, so they can add their point of view.

The debate shouldn't focus on "delta" versus "self-contained" profiles but rather on determining the appropriate level of detail to prevent issues arising from differing interpretations of the standard. Both profiles should be able to converge if we agree on a level of detail that leaves no room for interpretations.

Reply:
I commited https://github.com/camaraproject/IdentityAndConsentManagement/pull/121/commits/89311752fb43c761762869de818fbe719448afe7 addressing the missing scope value openid.

This document exists to improve interoperability. Defining the AZ's behavior in cases where OIDC says "unspecified" fits perfectly. If ICM does not agree with returning BAD REQUEST if openid is missing then please suggest alternative text.

Reply:
Thanks for pointing out the interoperability issues resp acr_values. I hope this commit improves interoperability for Carama
https://github.com/camaraproject/IdentityAndConsentManagement/pull/121/commits/99a44743e5f4f24679b6155177aadd566e7d6789